src/community/ogcapi/ogcapi-features/src/main/java/org/geoserver/ogcapi/v1/features/CQL2Conformance.java

```diff
@@ -0,0 +1,244 @@
+package org.geoserver.ogcapi.v1.features;
+
+import static org.geoserver.ogcapi.APIConformance.Level.DRAFT_STANDARD;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.geoserver.ogcapi.APIConformance;
+import org.geoserver.ogcapi.ConformanceClass;
+import org.geoserver.ogcapi.ConformanceInfo;
+import org.geoserver.wfs.WFSInfo;
+
+/** CQL2 Configuration for FeatureService. */
+public class CQL2Conformance extends ConformanceInfo<WFSInfo> {
+    public static final String METADATA_KEY = "cql2";
+    /** CQL Text conformance. */
+    public static final APIConformance CQL2_TEXT = new APIConformance(ConformanceClass.CQL2_TEXT);
+
+    /** CQL JSON conformance - not implemented yet (very different from the binding we have) */
+    public static final APIConformance CQL2_JSON =
+            new APIConformance(ConformanceClass.CQL2_JSON, DRAFT_STANDARD);
+
+    // CQL is optional
+    public static final APIConformance CQL2_ADVANCED =
+            new APIConformance(ConformanceClass.CQL2_ADVANCED);
+    public static final APIConformance CQL2_ARITHMETIC =
+            new APIConformance(ConformanceClass.CQL2_ARITHMETIC);
+    public static final APIConformance CQL2_BASIC = new APIConformance(ConformanceClass.CQL2_BASIC);
+    public static final APIConformance CQL2_BASIC_SPATIAL =
+            new APIConformance(ConformanceClass.CQL2_BASIC_SPATIAL);
+    public static final APIConformance CQL2_FUNCTIONS =
+            new APIConformance(ConformanceClass.CQL2_FUNCTIONS);
+
+    /** CQL2_TEMPORAL excluded for now, no support for all operators. */
+    public static final APIConformance CQL2_TEMPORAL =
+            new APIConformance(ConformanceClass.CQL2_TEMPORAL);
+
+    /** CQL2_ARRAY excluded, no support for array operations now. */
+    public static final APIConformance CQL2_ARRAY =
+            new APIConformance("http://www.opengis.net/spec/ogcapi-features-3/1.0/req/arrays");
+
+    public static final APIConformance CQL2_PROPERTY_PROPERTY =
+            new APIConformance(ConformanceClass.CQL2_PROPERTY_PROPERTY);
+    public static final APIConformance CQL2_SPATIAL =
+            new APIConformance(ConformanceClass.CQL2_SPATIAL);
+
+    // CQL2 formats
+    private Boolean json;
+    private Boolean text;
+
+    // CQL functionality
+    private Boolean advanced;
+    private Boolean arithmetic;
+    private Boolean basic;
+    private Boolean basicSpatial;
+    private Boolean functions;
+    private Boolean propertyProperty;
+    private Boolean spatial;
+
+    public CQL2Conformance() {}
+
+    /**
+     * Obtain CQL2Conformance configuration for WFSInfo.
+     *
+     * <p>Uses configuration stored in metadata map, or creates default if needed.
+     *
+     * @param wfsInfo WFSService configuration
+     * @return CQL2 configuration
+     */
+    public static CQL2Conformance configuration(WFSInfo wfsInfo) {
+        synchronized (wfsInfo) {
```

Comment:
Not sure this is going to do anything... the WFSInfo you have here should be a modification proxy created when the WFSInfo was requested (which must happen on a per request basis, these objects cannot be stored). In other words, synchronization should not be needed at all, if WFSInfo has been obtained properly (as said, dynamically in the current request, never stored as a field).